app/services/credit_notes/create_service.rb

```diff
@@ -23,6 +23,7 @@ def call
       return result.not_found_failure!(resource: "invoice") unless invoice
       return result.forbidden_failure! unless should_create_credit_note?
       return result.not_allowed_failure!(code: "invalid_type_or_status") unless valid_type_or_status?
+      return result.single_validation_failure!(field: :reason, error_code: "value_is_invalid") if invalid_reason?
```

Comment:
This is to continue to return the same error as before? instead of a "record_validation" error?

Reply:
What happens is that setting an invalid enum will raise `ArgumentError` instead of an adding a validation error when calling `.valid?` :

```ruby
lago-api(dev)> CreditNote.new(reason:'draft')
(lago-api):2:in '<main>': 'draft' is not a valid reason (ArgumentError)

          raise ArgumentError, "'#{value}' is not a valid #{name}"
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

My guess is that the initial code was introduced to catch invalid `reason` being provided. So what I did was to keep the same behavior for `reason` but allow other `ArgumentError` to bubble up.

Does that make sense ?

Reply:
I've updated it based on Vincent's comment.